### DISCUSSION TITLE
Execution order and environment

### DISCUSSION THREAD

**Discussion by erossini** [grate-devs/grate#338] [Q&A]
I want to organize the scripts for creating the database structure running the script in a particular order and based on the environment.

What do you recommend? I have some scripts that run in dev and test but not in production. Can I create the files like

    0001_environment1_environment2_scriptname.sql

when `environment1` is dev and `environment2` is test? Sometimes I have script for all environment. So, can I have a file like

    0002_scriptname.sql

that is executed after _0001_ for all environment?

**Comment by wokket** [comment]
G'day mate,

There's some docs on environment specific scripts here: https://erikbra.github.io/grate/environment-scripts/

tldr; is that yes, you can have scripts that only run in `dev` and `test` by naming the file per those docs.

**Comment by erossini** [comment]
Thank you for your answer. I have one question. In the filename, do I have to specified `ENV`? For example, in the [document](https://erikbra.github.io/grate/environment-scripts) you have

    LOCAL.permissionfile.env.sql
    AddTestData.Env.TEST.UAT.sql

I don't understand if this is mandatory or it is only an example?

**Comment by wokket** [comment]
Per the docs, when we see `.env.` in the file name, that triggers us to look for the passed in environment name (the `--env` value) in order to determine whether it should be run or not.

**Reply by erossini** [reply]
So, I have to add `.env.` in my files to trigger the environment. So, my name must be

    0001_environment1_environment2_.env.scriptname.sql

Without `.env.` the application doesn't filter for the environment. Is it correct?

**Reply by wokket** [reply]
Correct
